lib/address_geocoder.rb

```diff
@@ -150,13 +152,13 @@ def get_final_url(level_of_search)
   end
 
   def get_format_levels
-    levels  = [1,5]
-    levels += [2,6] if has_valid_city?
-    levels += [3,7] if has_valid_state?
+    levels  = [CycleWithPostalCode[:all], CycleWithoutPostalCode[:all]]
+    levels += [CycleWithPostalCode[:remove_street], CycleWithoutPostalCode[:remove_street]] if has_valid_city?
+    levels += [CycleWithPostalCode[:remove_city], CycleWithoutPostalCode[:remove_city]] if has_valid_state?
     if not_valid_postal_code?
-      levels  -= [1,2,3]
+      levels  -= CycleWithPostalCode.values
```

Comment:
Doesn't really matter, cause if you look a little further down you see that you only add the 4 to the array in the `else` of that `if`.